### PR TITLE
ensure core package customizations are processed

### PR DESCRIPTION
**Pull request by nabuskey**
fixes: #298

One thing to note when overwriting argocd CMs is that you need to have the labels on the CM. Otherwise, ArgoCD will complain about not finding `argocd-cm` CM.

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: argocd-cm
  labels: # these labels need to be present.
    app.kubernetes.io/name: argocd-cm 
    app.kubernetes.io/part-of: argocd
data:
  application.resourceTrackingMethod: annotation
  resource.exclusions: |
    - kinds:
        - ProviderConfigUsage
      apiGroups:
        - "*"
  timeout.reconciliation: 10s
```

@csantanapr ^^

### CODE REVIEW

File: pkg/controllers/resources/idpbuilder.cnoe.io_localbuilds.yaml [hunk 1]
```diff
@@ -64,23 +64,22 @@ spec:
...
+                    type: object
+                  packageCustomization:
+                    additionalProperties:
+                      description: PackageCustomization defines how packages are customized
+                      properties:
+                        filePath:
+                          description: FilePath is the absolute file path to a YAML
+                            file that contains Kubernetes manifests.
+                          type: string
+                        name:
```
Comment (blakeromano) [inline pkg/controllers/resources/idpbuilder.cnoe.io_localbuilds.yaml:76]: Can we make this an enum?

File: pkg/controllers/resources/idpbuilder.cnoe.io_localbuilds.yaml [hunk 2]
```diff
@@ -64,23 +64,22 @@ spec:
...
-                          - name
-                          type: object
-                        type: object
+                    type: object
+                  packageCustomization:
+                    additionalProperties:
+                      description: PackageCustomization defines how packages are customized
+                      properties:
+                        filePath:
+                          description: FilePath is the absolute file path to a YAML
```
Comment (blakeromano) [inline pkg/controllers/resources/idpbuilder.cnoe.io_localbuilds.yaml:73]: is the plan to not support helm or kustomize?

**Review by blakeromano — COMMENTED** [review]
I also don't really see much documentation around how this feature is supposed to be invoked/used...

Is this a CLI argument to `idpbuilder create`?

**Comment by csantanapr** [comment]
Good catch on the labels @nabuskey

**Review by csantanapr — APPROVED** [review]
/lgtm

**Review by nimakaviani — APPROVED** [review]

**Comment by nabuskey** [comment]
> I also don't really see much documentation around how this feature is supposed to be invoked/used...
> 
> Is this a CLI argument to `idpbuilder create`?

We do need to do a better job of documenting these.
See: https://github.com/cnoe-io/idpbuilder/pull/180

**Pull request merged**